### PR TITLE
Adjustments to README

### PR DESCRIPTION
**Pull request by jasonblais**
1. Add links to signup to Cloud or deploy on-premises
2. Add "learn more" resources for DevSecOps, Incident Resolution and IT Service Desk use cases. This is to gauge what use cases people visiting the open-source project may have when visiting our repo

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,9 +2,17 @@
...
 [Mattermost](https://mattermost.com) is an open source platform for secure collaboration across the entire software development lifecycle. This repo is the primary source for core development on the Mattermost platform; it's written in Go and React and runs as a single Linux binary with MySQL or PostgreSQL. A new compiled version is released under an MIT license every month on the 16th.
 
+[Use it for free in Mattermost Cloud](https://mattermost.com/sign-up/) or [deploy on-premises](https://mattermost.com/deploy/).
+
 <img width="1006" alt="mattermost-hero" src="https://user-images.githubusercontent.com/7205829/136107976-7a894c9e-290a-490d-8501-e5fdbfc3785a.png">
 
-Useful resources:
+Learn more about the following use cases with Mattermost:
+
+- [DevSecOps](https://mattermost.com/solutions/use-cases/devops/?source=github-mattermost-server-readme)
```
Comment (chenilim) [inline README.md:11]: Should we use `utm_source=...` for the query strings?

File: README.md [hunk 2]
```diff
@@ -2,9 +2,17 @@
...
+- [Incident Resolution](https://mattermost.com/solutions/use-cases/incident-resolution/?source=github-mattermost-server-readme)
```
Comment (chenilim) [inline README.md:12]: I think these three look good to start with, and we can iterate. E.g. not sure if it should be "Incident resolution" vs. "incident response" or "incident management". But this is consistent with the website.

File: README.md [hunk 3]
```diff
@@ -2,9 +2,17 @@
 
 [Mattermost](https://mattermost.com) is an open source platform for secure collaboration across the entire software development lifecycle. This repo is the primary source for core development on the Mattermost platform; it's written in Go and React and runs as a single Linux binary with MySQL or PostgreSQL. A new compiled version is released under an MIT license every month on the 16th.
 
+[Use it for free in Mattermost Cloud](https://mattermost.com/sign-up/) or [deploy on-premises](https://mattermost.com/deploy/).
```
Comment (chenilim) [inline README.md:5]: Should we add utm_source tags to these too?

File: README.md (hunk 1 above)
Comment (jasonblais) [reply README.md:11]: Yes, sorry, changed to use `utm_source` instead of `source`

File: README.md (hunk 2 above)
Comment (jasonblais) [reply README.md:12]: Agree, I just used what we have on our website to start

File: README.md (hunk 3 above)
Comment (jasonblais) [reply README.md]: +1, good call. Added.

File: README.md [hunk 4]
```diff
@@ -2,9 +2,17 @@
...
+[Use it for free in Mattermost Cloud](https://mattermost.com/sign-up/?utm_source=github-mattermost-server-readme) or [deploy on-premises](https://mattermost.com/deploy/?utm_source=github-mattermost-server-readme).
+
 <img width="1006" alt="mattermost-hero" src="https://user-images.githubusercontent.com/7205829/136107976-7a894c9e-290a-490d-8501-e5fdbfc3785a.png">
 
-Useful resources:
+Learn more about the following use cases with Mattermost:
+
+- [DevSecOps](https://mattermost.com/solutions/use-cases/devops/?utm_source=github-mattermost-server-readme)
+- [Incident Resolution](https://mattermost.com/solutions/use-cases/incident-resolution/?utm_source=github-mattermost-server-readme)
+- [IT Service Desk](https://mattermost.com/solutions/use-cases/it-service-desk//?utm_source=github-mattermost-server-readme)
```
Comment (chenilim) [inline README.md:13]: Minor thing: double slash `//` at the end of the path.

**Review by chenilim — APPROVED** [review]
Looks good! Just an extra slash at the end of one path, but that shouldn't affect anything.

File: README.md (hunk 4 above)
Comment (jasonblais) [reply README.md]: Thanks for your attention to detail, fixed 🙏

**Review by amyblais — APPROVED** [review]

**Comment by jasonblais** [comment]
Merging with two approved reviews

**Pull request merged**